### PR TITLE
Listen javascript function from flutter

### PR DESCRIPTION
**Pull request by rinlv**
See an example [here](https://github.com/rinlv/flutter_webview_plugin/blob/master/example/lib/main.dart)

### CODE REVIEW

**Comment by samuelsarix** [comment]
Thank You, needed this

**Comment by charafau** [comment]
Thanks for PR, give me some time to review this. Just looking at title of this - https://github.com/fluttercommunity/flutter_webview_plugin/pull/457 what's difference between the two ?

**Comment by rinlv** [comment]
@charafau This #457 use fixed channel 'Android' to listen to JS function for Android version, my solutions help using the dynamic channel for both Android and iOS. More, I'm following this [webview_flutter](https://pub.dev/packages/webview_flutter) to make this PR. It's mean just call 
<Channel>.postMessage(data) to listen to JS function for both Android and iOS, not complicated like [this](https://github.com/OPY-bbt/flutter_webview_plugin/blob/message/example/html/index.html) example of #457

**Comment by ivanasen** [comment]
@rinlv I've made a few changes to dart part, namely adding the whole Javascript channel abstraction, which removes the need to do a switch on the channel for example when a message is received. I've made a pull request on your repo [here](https://github.com/rinlv/flutter_webview_plugin/pull/1)

**Comment by PhanPirang** [comment]
@rinlv @ivanasen I clone your repo, after i click on Display Datetime button, I got "Uncaught ReferenceError: Print is not defined", what's the problem with Channel?

**Comment by charafau** [comment]
I will make review and check code on Monday 👍

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -181,7 +181,7 @@
 				TargetAttributes = {
```
Comment (charafau) [inline example/ios/Runner.xcodeproj/project.pbxproj]: I think you should revert changes for this file

File: example/lib/main.dart [hunk 2]
```diff
@@ -7,7 +7,16 @@ import 'package:flutter_webview_plugin/flutter_webview_plugin.dart';
 const kAndroidUserAgent =
     'Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Mobile Safari/537.36';
 
-String selectedUrl = 'https://flutter.io';
+String selectedUrl = 'https://damp-coast-35782.herokuapp.com';
```
Comment (charafau) [inline example/lib/main.dart]: please revert this link. if you want to do that, maybe add button with another example

File: android/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
           xmlns:tools="http://schemas.android.com/tools"
           package="com.flutter_webview_plugin">
-    <uses-sdk android:minSdkVersion="16" />
+
```
Comment (charafau) [inline android/src/main/AndroidManifest.xml]: I think there was PR for this to add min sdk, why was it removed?

Comment (rinlv) [reply android/src/main/AndroidManifest.xml]: I removed by the suggestion of IDE, I will revert it

**Comment by rinlv** [comment]
@charafau Please check then merge this pull

**Comment by charafau** [comment]
@rinlv thanks for commits! I will do that after weekend, got urgent plans

**Comment by alereisan** [comment]
@charafau is this already available? I am getting Error: No named parameter with the name 'javascriptChannels' on version 0.3.8

**Comment by rjsajnani** [comment]
Even i faced this same issue so i added this to my code base and it worked fine 

`dependencies:
   flutter_webview_plugin:
     git:
       url: https://github.com/fluttercommunity/flutter_webview_plugin.git`

**Comment by mzF414** [comment]
How to use the 'JavascriptChannel' ?

**Comment by mzF414** [comment]
OK, I got it 
[https://github.com/flutter/plugins/blob/master/packages/webview_flutter/example/lib/main.dart](url)

**Pull request merged**